scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua

```diff
@@ -0,0 +1,45 @@
+-----------------------------------
+-- Area: Bastok Markets [S]
+--  NPC: Kevan
+-- !pos -308.590 -012.000 -094.227 189
+-----------------------------------
+local ID = require("scripts/zones/Bastok_Markets_[S]/IDs")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    local itemID = trade:getItemId()
+    local itemKeyItemMapping = {
+        [2686] = xi.ki.THE_WORDS_OF_DONHU_I,
+        [2687] = xi.ki.THE_WORDS_OF_DONHU_II,
+        [2688] = xi.ki.THE_WORDS_OF_DONHU_III,
+        [2689] = xi.ki.THE_WORDS_OF_DONHU_IV,
+        [2690] = xi.ki.THE_WORDS_OF_DONHU_V,
+        [2691] = xi.ki.THE_WORDS_OF_DONHU_VI,
+        [2692] = xi.ki.THE_WORDS_OF_DONHU_VII,
+        [2693] = xi.ki.THE_WORDS_OF_DONHU_VIII
+    }
+    -- Make sure we're only trading 1 Strongbox at a time.
+    if trade:hasItemQty(itemID, 1) and trade:getItemCount() == 1 and itemKeyItemMapping[itemID] then
+        player:tradeComplete()
+        local keyItem = itemKeyItemMapping[itemID]
+        npc:showText(npc, 13549)
+        player:addKeyItem(keyItem)
+        player:messageSpecial(ID.text.KEYITEM_OBTAINED, keyItem)
```

Comment:
`npcUtil.giveKeyItem(player, keyItem)` handles the message.
So there would be no need for the message.

Also, we could add the message used in `npc:showText(npc, 13549)` in the IDs file to not use a magic number/raw ID

Reply:
Resolved the first half, will resolve second half once I am 100% sure these are even the right events.  I wasn't able to farm any of these items and pull the event ID on retail yet, planning on doing that tomorrow because I don't think the events in the logs I used are still up to date per comment.